model/task_lifecycle.go

```diff
@@ -1429,17 +1434,16 @@ func UpdateVersionAndPatchStatusForBuilds(buildIds []string) error {
 		if !buildStatusChanged {
 			continue
 		}
-
-		versionsToUpdate[build.Version] = build.Id
+		versionsToUpdate[build.Version] = append(versionsToUpdate[build.Version], build.Id)
 		bvMap[build.Id] = build.BuildVariant
 	}
-	for versionId, buildId := range versionsToUpdate {
+	for versionId, buildList := range versionsToUpdate {
 		buildVersion, err := VersionFindOneId(versionId)
 		if err != nil {
-			return errors.Wrapf(err, "getting version '%s' for build '%s'", versionId, buildId)
+			return errors.Wrapf(err, "getting version '%s' for build '%s'", versionId, buildList[0])
```

Comment:
could just returning the first element of the buildList be misleading sometimes?

Reply:
Sure, I'm using a `strings.Join` now instead.